gammapy/irf/psf_table.py

```diff
@@ -567,12 +568,16 @@ def containment_radius(self, energies, fraction, interp_kwargs=None):
         rad : `~astropy.units.Quantity`
             Containment radius in deg
         """
-        # TODO: figure out if there's a more efficient implementation to support
-        # arrays of energy
-        energies = np.atleast_1d(energies)
-        psfs = [self.table_psf_at_energy(energy, interp_kwargs) for energy in energies]
-        rad = [psf.containment_radius(fraction) for psf in psfs]
-        return Quantity(rad)
+        # oversample for better precision
+        rad = np.linspace(0 * u.deg, self.rad.max(), 10 * len(self.rad))
+
+        psf_value = self.evaluate(energy=energies, rad=rad).to_value("deg-2")
+        containment = cumtrapz(y=rad * psf_value, x=rad.to_value("deg"), axis=1)
+
+        with np.errstate(divide="ignore", invalid="ignore"):
```

Comment:
Should the renormalization be the default behavior? Or should it be done here, instead of `__init__` for instance?

Reply:
I think this is basically an open question. I guess ideally the PSF would be normalized already on the DL3 data level and only re-normalized in Gammapy, when the normalization changed because of some computation (such as broadening or clipping the PSF). Right now PSFs are not necessarily normalized and we need to do it here to get the correct containment.